[Docs platform] Ruby version

We currently build docs using Ruby 2.7 which works just fine

```
➜  lakeFS git:(master) ruby -v
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [-darwin22]
➜  lakeFS git:(master) make docs-serve
cd docs; bundle exec jekyll serve
Configuration file: /Users/rmoff/git/lakeFS/docs/_config.yml
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
            Source: /Users/rmoff/git/lakeFS/docs
       Destination: /Users/rmoff/git/lakeFS/docs/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
      Remote Theme: Using theme pmarsceill/just-the-docs
   GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
                    done in 51.17 seconds.
 Auto-regeneration: enabled for '/Users/rmoff/git/lakeFS/docs'
    Server address: http://127.0.0.1:4000
  Server running... press ctrl-c to stop.
```

If/when we move to Ruby 3 we need to add the `webrick` gem otherwise the build fails

```
➜  lakeFS git:(master) ✗ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]
➜  lakeFS git:(master) ✗ make docs-serve
cd docs; bundle exec jekyll serve
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Configuration file: /Users/rmoff/git/lakeFS/docs/_config.yml
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
            Source: /Users/rmoff/git/lakeFS/docs
       Destination: /Users/rmoff/git/lakeFS/docs/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
      Remote Theme: Using theme pmarsceill/just-the-docs
   GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
                    done in 7.513 seconds.
 Auto-regeneration: enabled for '/Users/rmoff/git/lakeFS/docs'
bundler: failed to load command: jekyll (/Users/rmoff/.gem/ruby/3.1.3/bin/jekyll)
LoadError: cannot load such file -- webrick
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve/servlet.rb:3:in `require'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve/servlet.rb:3:in `<top (required)>'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:184:in `require_relative'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:184:in `setup'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:102:in `process'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `block in start'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `each'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `start'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:75:in `block (2 levels) in init_with_program'
  /Users/rmoff/.gem/ruby/3.1.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
  /Users/rmoff/.gem/ruby/3.1.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
  /Users/rmoff/.gem/ruby/3.1.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
  /Users/rmoff/.gem/ruby/3.1.3/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
  /Users/rmoff/.gem/ruby/3.1.3/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
  /Users/rmoff/.gem/ruby/3.1.3/gems/jekyll-3.9.3/exe/jekyll:15:in `<top (required)>'
  /Users/rmoff/.gem/ruby/3.1.3/bin/jekyll:25:in `load'
  /Users/rmoff/.gem/ruby/3.1.3/bin/jekyll:25:in `<top (required)>'
make: *** [docs-serve] Error 1
```

--- a/.github/workflows/check-ui-links.yaml
+++ b/.github/workflows/check-ui-links.yaml
@@ -19,7 +19,7 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           working-directory: docs
-          ruby-version: '2.7'
+          ruby-version: '3.2'
           bundler-cache: true 
 
       - name: Build latest

--- a/.github/workflows/docs-latest.yaml
+++ b/.github/workflows/docs-latest.yaml
@@ -19,7 +19,7 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           working-directory: docs
-          ruby-version: '2.7'
+          ruby-version: '3.2'
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
 
       - name: Build latest

--- a/.github/workflows/docs-pr.yaml
+++ b/.github/workflows/docs-pr.yaml
@@ -20,7 +20,7 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           working-directory: docs
-          ruby-version: '2.7'
+          ruby-version: '3.2'
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
 
       - name: Build latest

--- a/.github/workflows/docs-release.yaml
+++ b/.github/workflows/docs-release.yaml
@@ -20,7 +20,7 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           working-directory: docs
-          ruby-version: '2.7'
+          ruby-version: '3.2'
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
 
       - name: Extract version

--- a/docs/Gemfile
+++ b/docs/Gemfile
@@ -1,3 +1,4 @@
 source 'https://rubygems.org'
 gem 'github-pages', group: :jekyll_plugins
 gem 'just-the-docs'
+gem "webrick", "~> 1.8"

